src/test/java/spoon/test/variable/AccessTest.java

```diff
@@ -0,0 +1,142 @@
+
+		final List<CtVariableAccess<?>> variablesAccess =
+				Query.getElements(factory,
+								  new AbstractFilter<CtVariableAccess<?>>(CtVariableAccess.class) {
+									  @Override
+									  public boolean matches(CtVariableAccess<?> element) {
+										  return super.matches(element);
+									  }
+								  });
+
+		assertEquals(variablesRead.size() + variablesWrite.size(), variablesAccess.size());
+	}
+
+	@Test
+	public void testCanVisitFieldAccessAndSubClasses() throws Exception {
+		final Factory factory = TestUtils.build(FieldAccessSample.class);
+
+		final List<CtFieldRead<?>> fieldsRead =
+				Query.getElements(factory,
+								  new AbstractFilter<CtFieldRead<?>>(CtFieldRead.class) {
+									  @Override
+									  public boolean matches(CtFieldRead<?> element) {
+										  return super.matches(element);
+									  }
+								  });
+
+		assertEquals(2, fieldsRead.size());
+
+		final List<CtFieldWrite<?>> fieldsWrite =
+				Query.getElements(factory,
+								  new AbstractFilter<CtFieldWrite<?>>(CtFieldWrite.class) {
+									  @Override
+									  public boolean matches(CtFieldWrite<?> element) {
+										  return super.matches(element);
+									  }
+								  });
+
+		assertEquals(3, fieldsWrite.size());
+
+		final List<CtFieldAccess<?>> fieldsAccess =
+				Query.getElements(factory,
+								  new AbstractFilter<CtFieldAccess<?>>(CtFieldAccess.class) {
+									  @Override
+									  public boolean matches(CtFieldAccess<?> element) {
+										  return super.matches(element);
+									  }
+								  });
+
+		assertEquals(fieldsRead.size() + fieldsWrite.size(), fieldsAccess.size());
+	}
+
+	@Test
+	public void testCanVisitArrayAccessAndSubClasses() throws Exception {
+		final Factory factory = TestUtils.build(ArrayAccessSample.class);
+
+		final List<CtArrayRead<?>> arraysRead =
+				Query.getElements(factory,
+								  new AbstractFilter<CtArrayRead<?>>(CtArrayRead.class) {
+									  @Override
+									  public boolean matches(CtArrayRead<?> element) {
+										  return super.matches(element);
+									  }
+								  });
+
+		assertEquals(1, arraysRead.size());
+
+		final List<CtArrayWrite<?>> arraysWrite =
+				Query.getElements(factory,
+								  new AbstractFilter<CtArrayWrite<?>>(CtArrayWrite.class) {
+									  @Override
+									  public boolean matches(CtArrayWrite<?> element) {
+										  return super.matches(element);
+									  }
+								  });
+
+		assertEquals(1, arraysWrite.size());
+
+		final List<CtArrayAccess<?, CtExpression<?>>> arraysAccess =
+				Query.getElements(factory,
+								  new AbstractFilter<CtArrayAccess<?, CtExpression<?>>>(
+										  CtArrayAccess.class) {
+									  @Override
+									  public boolean matches(
+											  CtArrayAccess<?, CtExpression<?>> element) {
+										  return super.matches(element);
+									  }
+								  });
+
+		assertEquals(arraysRead.size() + arraysWrite.size(), arraysAccess.size());
```

Comment:
I disagree. The size of `CtArrayAccess` (the same for `CtVariableAccess` and `CtFieldAccess`) must be the sum of `CtArrayRead` and `CtArrayWrite`.